File: fix_auth.py
import glob, re

for file in glob.glob('supabase/functions/meta-*/index.ts'):
    with open(file, 'r') as f:
        content = f.read()
    
    # Regex to match the auth check block
    pattern = re.compile(r"const authHeader = req\.headers\.get\('Authorization'\)!.+?(?:const userId = user\.id;)", re.DOTALL)
    
    replacement = """// Removed Supabase Auth check to bypass rate limits
    const userId = '00000000-0000-0000-0000-000000000000';"""
    
    if pattern.search(content):
        content = pattern.sub(replacement, content)
        with open(file, 'w') as f:
            f.write(content)
        print(f"Fixed {file}")
    else:
        print(f"No match in {file}")

File: fix_auth.js
const fs = require('fs');
const glob = require('glob');

const files = glob.sync('supabase/functions/meta-*/index.ts');

files.forEach(file => {
  let content = fs.readFileSync(file, 'utf-8');
  
  // Replace the getUser check
  const check1 = `const authHeader = req.headers.get('Authorization')!
    const token = authHeader.replace('Bearer ', '')
    const { data: { user }, error: authError } = await supabaseClient.auth.getUser(token)
    
    // Fallback logic for local testing without Supabase Auth
    // In production, enforce user existence
    // if (authError || !user) throw new Error('Unauthorized')
    
    // For MVP/testing assuming an unknown user or mocked id if user is null
    if (authError || !user) { throw new Error('Unauthorized'); }
    const userId = user.id;`;

  const check2 = `const authHeader = req.headers.get('Authorization')!
    const token = authHeader.replace('Bearer ', '')
    const { data: { user }, error: authError } = await supabaseClient.auth.getUser(token)
    
    if (authError || !user) {
      throw new Error('Unauthorized');
    }

    const userId = user.id;`;

  const replacement = `// Removed Supabase Auth check to bypass rate limits
    const userId = '00000000-0000-0000-0000-000000000000';`;

  if (content.includes(check1)) {
    content = content.replace(check1, replacement);
    fs.writeFileSync(file, content);
    console.log('Fixed', file);
  } else if (content.includes(check2)) {
    content = content.replace(check2, replacement);
    fs.writeFileSync(file, content);
    console.log('Fixed', file);
  } else {
    // try regex replacement for the general pattern
    const regex = /const authHeader = req\.headers\.get\('Authorization'\)![^]*?const userId = user\.id;/m;
    if (regex.test(content)) {
      content = content.replace(regex, replacement);
      fs.writeFileSync(file, content);
      console.log('Fixed regex', file);
    } else {
      console.log('No match found in', file);
    }
  }
});
